File: python/m20.function.py
#!/usr/bin/python

# sort function that returns a sorted array
def sort_array (array):
   return sorted(array)  # return list

# initialize initial array (list)
array = ["bob","ed","steve","ralph","joe","deb","kate"]
# output current list before calling function
print "Current names are: ", array

# initialize new array from output of function
result = sort_array(array)
print "Sorted names are:  ", result

File: gen_scripts/perl/m20.function.pl
#!/usr/bin/env perl
# create the function
sub sort_array {
   return join(", ", sort @_);  # return list
}

# initialize initial array (list)
my @array = ("bob", "ed", "steve", "ralph", "joe", "deb", "kate");
# output current list before calling function
print "Current names are: " . join(", ", @array) . "\n";

# call the function
my $result = sort_array(@array);

# output the result
print "Sorted names are: $result\n";
